frontend/src/components/core/MainMenu/MainMenu.tsx

```diff
@@ -308,59 +362,105 @@ function MainMenu(props: Props): ReactElement {
-    deployApp: {
-      onClick: onClickDeployApp,
-      label: "Deploy this app",
-    },
     saveSnapshot: {
       disabled: isServerDisconnected,
       onClick: props.shareCallback,
       label: "Save a snapshot",
     },
-    documentation: {
-      onClick: getOpenInWindowCallback(ONLINE_DOCS_URL),
-      label: "Documentation",
+    ...(!props.menuItems?.hideGetHelp && {
+      community: {
+        onClick: getOpenInWindowCallback(
+          props.menuItems?.getHelpUrl || COMMUNITY_URL
+        ),
+        label: "Get help",
+      },
+    }),
+    ...(!props.menuItems?.hideReportABug && {
+      report: {
+        onClick: getOpenInWindowCallback(
+          props.menuItems?.reportABugUrl || BUG_URL
+        ),
+        label: "Report a bug",
+      },
+    }),
+    settings: { onClick: props.settingsCallback, label: "Settings" },
+    about: { onClick: props.aboutCallback, label: "About" },
+  }
+
+  const coreDevMenuItems = {
+    DIVIDER: { isDivider: true },
+    deployApp: {
+      onClick: onClickDeployApp,
+      label: "Deploy this app",
     },
-    community: {
-      onClick: getOpenInWindowCallback(COMMUNITY_URL),
-      label: "Ask a question",
+    developerOptions: {
+      label: "Developer options",
+      noHighlight: true,
+      interactions: {},
+      styleProps: {
+        fontSize: "0.75rem",
+        margin: "-.5rem 0 0 0",
+        padding: ".25rem 0 .25rem 1.5rem",
+        pointerEvents: "none",
+      },
     },
-    report: {
-      onClick: getOpenInWindowCallback(BUG_URL),
-      label: "Report a bug",
+    clearCache: {
+      disabled: isServerDisconnected,
+      onClick: props.clearCacheCallback,
+      label: "Clear cache",
+      shortcut: "c",
     },
     s4t: {
       onClick: getOpenInWindowCallback(TEAMS_URL),
-      label: "Streamlit for Teams",
+      label: "Streamlit Cloud",
+    },
+    reportSt: {
+      onClick: getOpenInWindowCallback(BUG_URL),
+      label: "Report a Streamlit bug",
+    },
+    documentation: {
+      onClick: getOpenInWindowCallback(ONLINE_DOCS_URL),
+      label: "Visit Streamlit docs",
+    },
+    visitStForum: {
+      onClick: getOpenInWindowCallback(COMMUNITY_URL),
+      label: "Visit Streamlit forums",
+      styleProps: {
+        margin: "0 0 -.5rem 0",
+        padding: ".25rem 0 .25rem 1.5rem",
+      },
     },
-    settings: { onClick: props.settingsCallback, label: "Settings" },
-    about: { onClick: props.aboutCallback, label: "About" },
   }
 
-  const S4AMenuOptions = props.s4aMenuItems.map(item => {
+  const S4AMenuItems = props.s4aMenuItems.map(item => {
     if (item.type === "separator") {
-      return coreMenuOptions.DIVIDER
+      return coreMenuItems.DIVIDER
+    }
+
+    if (item.label === "Report a bug with this app") {
+      if (props.menuItems?.hideGetHelp) {
+        return null
+      }
+    }
+
+    if (item.label === "About Streamlit Cloud") {
+      if (props.menuItems?.aboutSectionMd !== "") {
+        return null
+      }
```

Comment:
This feels a bit hacky, but I noticed that the `IMenuItem` type can also have a `key` field, which I think may be usable for this? If so, it'd be a bit better to check against the key rather than the menuItem's label.

If `key` doesn't work the way I hope it does, then I think this is fine given that we don't have much of an alternative right now, but we'll want to talk to the cloud team about figuring out a better way to do this since this will break if the  labels ever change (which could easily happen due to some product changes without us realizing the implications on the core side of things)